### ISSUE TITLE
UDVC-Server.x64.exe has error 998, then exits with error 10054 on first packet sent

### ISSUE THREAD

**Issue by MarkCranness** [earthquake/UniversalDVC#4]
"_Visual Studio Remote Debugging UDVC-Server.x64.exe" -c -p 4018
Universal Dynamic Virtual Channel server application

[*] Setting up client socket
[*] Connected to: 127.0.0.1:4018
[*] Starting thread RsWc
[*] Starting thread RcWs
[*] Waiting for threads to exit...
[-] [RsWc] WriteFile()/WaitForSingleObject() error: 998
[-] [RsWc] send() failed with error 10054

error 998 happens a few seconds after starting.
error 10054 happens upon the first connect attempt.

UDVC-Server.x64 running on Windows Server 2012 R2 Standard
Addin on Windows 10 x64
(Software versions from the pre-built Releases here, and renamed with "_Visual Studio Remote Debugging" prefix.)

**Comment by earthquake** [comment]
Hello Mark,

This tool can only used for one connection/run. This means if the TCP connection was closed on either side the tool will exit and close down all communication channels. I believe what happened is, the TCP connection died/got reset on either side, so the send() function failed and the tool exited.

What exactly did you try to run over the tool?

Thanks,

**Comment by MarkCranness** [comment]
Ah yes, thanks.
Overnight I did wonder about multiple connections, and upon checking, the tool does open multiple connections (possibly even more than 1 at a time), so this is user error.

The tool I was trying to use is Visual Studio Remote Debugging, where Visual Studio editor allows you to attach to a process running on another host and debug it.
The endpoints are devenv.exe (Visual Studio) locally, and msvmon.exe (The remote debug helper) on the remote host.

I have got it working with SocksOverRDP and nmap's ncat.exe to tunnel thru the socks proxy.

SocksOverRDP must be based on your UniversalDVC, so thank you!
